commands/orchestrationcommands/stackcommands/preview.go

```diff
@@ -0,0 +1,160 @@
+package stackcommands
+
+import (
+	"errors"
+
+	"github.com/rackspace/rack/commandoptions"
+	"github.com/rackspace/rack/handler"
+	"github.com/rackspace/rack/internal/github.com/codegangsta/cli"
+	osStacks "github.com/rackspace/rack/internal/github.com/rackspace/gophercloud/openstack/orchestration/v1/stacks"
+	"github.com/rackspace/rack/util"
+)
+
+var preview = cli.Command{
+	Name:        "preview",
+	Usage:       util.Usage(commandPrefix, "preview", "[--name <stackName> | --stdin name] [--template-file <templateFile> | --template-url <templateURL>]"),
+	Description: "Preview a stack",
+	Action:      actionPreview,
+	Flags:       commandoptions.CommandFlags(flagsPreview, nil),
+	BashComplete: func(c *cli.Context) {
+		commandoptions.CompleteFlags(commandoptions.CommandFlags(flagsPreview, keysPreview))
+	},
+}
+
+func flagsPreview() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "[optional; required if `stdin` isn't provided] The stack name.",
+		},
+		cli.StringFlag{
+			Name:  "stdin",
+			Usage: "[optional; required if `name` isn't provided] The field being piped into STDIN. Valid values are: name.",
+		},
+		cli.StringFlag{
+			Name:  "template-file",
+			Usage: "[optional; required if `template-url` isn't provided] The path to template file.",
+		},
+		cli.StringFlag{
+			Name:  "template-url",
+			Usage: "[optional; required if `template-file` isn't provided] The url to the template.",
+		},
+		cli.StringFlag{
+			Name:  "environment-file",
+			Usage: "[optional] File containing environment for the stack",
```

Comment:
I was just reading below: does this need to be a URL? If so, we should specify that for all the `environment-file` flags.